### PR TITLE
enabled job and pod resource change via patch

### PR DESCRIPTION
**Pull request by pdgetrf**
(1) enabled patch resource for both job and pod (cpu, memory, etc.)
(2) added check against qos change as a result of (1). qos change is not allowed
(3) updated and added unit tests

tested through unit and e2e tests on a local environment.

### CODE REVIEW

File: pkg/apis/core/validation/validation_test.go [hunk 1]
```diff
@@ -8062,6 +8062,36 @@ func TestValidatePodUpdate(t *testing.T) {
...
+							Image: "foo:V2",
+							Resources: core.ResourceRequirements{
+								Limits: getResourceLimits("100m", "3Gi"),
+							},
+						},
+					},
+				},
+			},
+			"",
+			"memory change",
```
Comment (vinaykul) [inline pkg/apis/core/validation/validation_test.go:8093]: please try image name foo:V1 for old and new, and make this "memory only change"

File: pkg/apis/core/validation/validation_test.go [hunk 2]
```diff
@@ -8089,9 +8119,105 @@ func TestValidatePodUpdate(t *testing.T) {
 					},
 				},
 			},
-			"spec: Forbidden: pod updates may not change fields",
+			"",
```
Comment (vinaykul) [inline pkg/apis/core/validation/validation_test.go:8122]: we can have this test with this change, and add another test without varying image, "cpu only change"

File: pkg/apis/core/validation/validation_test.go [hunk 3]
```diff
@@ -8089,9 +8119,105 @@ func TestValidatePodUpdate(t *testing.T) {
...
+						},
+					},
+				},
+			},
+			core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Image: "foo:V2",
```
Comment (vinaykul) [inline pkg/apis/core/validation/validation_test.go]: please try foo:V1

File: pkg/apis/core/validation/validation_test.go [hunk 4]
```diff
@@ -8089,9 +8119,105 @@ func TestValidatePodUpdate(t *testing.T) {
...
```
Comment (vinaykul) [inline pkg/apis/core/validation/validation_test.go]: same here - foo:V1

File: pkg/apis/core/validation/validation_test.go [hunk 5]
```diff
@@ -8089,9 +8119,105 @@ func TestValidatePodUpdate(t *testing.T) {
...
```
Comment (vinaykul) [inline pkg/apis/core/validation/validation_test.go]: foo:V1

**Review by vinaykul — CHANGES_REQUESTED** [review]
Please see comments below. LGTM after those are fixed.

File: pkg/apis/core/validation/validation_test.go (hunk 1 above)
Comment (pdgetrf) [reply pkg/apis/core/validation/validation_test.go:8093]: fixed

File: pkg/apis/core/validation/validation_test.go (hunk 2 above)
Comment (pdgetrf) [reply pkg/apis/core/validation/validation_test.go:8122]: there's already a test for image change. changed this to cpu only with the same image.

File: pkg/apis/core/validation/validation_test.go (hunk 5 above)
Comment (pdgetrf) [reply pkg/apis/core/validation/validation_test.go]: fixed

File: pkg/apis/core/validation/validation_test.go (hunk 4 above)
Comment (pdgetrf) [reply pkg/apis/core/validation/validation_test.go]: fixed

File: pkg/apis/core/validation/validation_test.go (hunk 3 above)
Comment (pdgetrf) [reply pkg/apis/core/validation/validation_test.go]: fixed

**Review by vinaykul — APPROVED** [review]

**Pull request merged**